File: backend/Listify/Listify.Api/DTOs/Listing/CreateListingDto.cs
using Listify.Application.DTOs;
using Listify.Domain;

namespace Listify.Api.DTOs.Listing;

public class CreateListingDto
{
    public string Title { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public LocationDto Location { get; set; }
    public int CategoryId { get; set; }
    
    public IEnumerable<ListingAttributeDto> ListingAttributeDtos { get; set; } = new List<ListingAttributeDto>();
    public IEnumerable<ListingImageDto> ListingImageDtos { get; set; } = new List<ListingImageDto>();
}

File: frontend/Listify.Frontend/src/DTOs/Listing/CreateListingDto.ts
export type CreateListingAttributeDto = {
  CategoryAttributeValueId: number
}

export type CreateListingImageDto = {
  Url: string
  Order: number
  PublicId: string
}

export type CreateListingLocationDto = {
  Name: string
  Ref: string
  Area: string
}

export type CreateListingDto = {
  Title: string
  Description: string
  Price: number
  Location: CreateListingLocationDto
  CategoryId: number
  ListingAttributeDtos: CreateListingAttributeDto[]
  ListingImageDtos: CreateListingImageDto[]
}
